src/mon/HealthMonitor.h

```diff
@@ -45,9 +45,10 @@ class HealthMonitor : public QuorumService
   virtual bool service_dispatch(MonOpRequestRef op);
 
   virtual void start_epoch() {
+    epoch_t epoch = get_epoch();
     for (map<int,HealthService*>::iterator it = services.begin();
          it != services.end(); ++it) {
-      it->second->start(get_epoch());
```

Comment:
@tchaikov  Yes sir.That is true.But the modify here is really save time.I write a very simple test like below:

[root@localhost test]# cat test1.cc 
#include <unistd.h>
#include <fcntl.h>
#include <errno.h>
#include <string.h>
#include <stdlib.h>
#include <string>
#include <iostream>
#include <stdio.h>
#include <sys/stat.h>
#include <sys/types.h>
using namespace std;
#define TEST_MAX 10


unsigned long long gettickcount(void) {
  struct timespec ts;
  if (clock_gettime(CLOCK_MONOTONIC, &ts) != 0) {
    return 0;
  }
  return (ts.tv_nsec) + ts.tv_sec * 1000000000;
}

class ctest
{
public:
      void setvalue(int i) {ival = i;};
      int getvalue() {return ival;};
      void test1()
      {
         for(int i = 0; i < TEST_MAX; i++)
         {
             int a =  getvalue();
         }
      };

      void test2()
      {
         int b = getvalue();
         for(int i = 0; i < TEST_MAX; i++)
         {
             int a = b;
         }
      };

private:
     int ival;

};





int main()
{
  ctest  t;
  t.setvalue(5);
  unsigned long long t1, t2;
  t1 = gettickcount();
  t.test1();
  t1 = gettickcount() - t1;
  t2 = gettickcount();
  t.test2();
  t2 = gettickcount() - t2;
  printf("v1: %llu ns \n", t1 );
  printf("v2: %llu ns\n", t2 );
  return 0;
}
[root@localhost test]# g++ -o code test1.cc -O2 --std=c++11 -lrt
[root@localhost test]# ./code 
v1: 313 ns 
v2: 116 ns
[root@localhost test]#